collectors/ciscoamp/utils.js

```diff
@@ -22,39 +20,30 @@ function getAPILogs(baseUrl, authorization, apiUrl, state, accumulator, maxPages
                         "authorization": `Basic ${authorization}`
                     },
                     resolveWithFullResponse: true
-                }).then(({ headers, body }) => {
+                }).then(({ body }) => {
                     pageCount++;
-                    if (parseInt(headers['x-ratelimit-remaining']) < 200) {
-                        resetSeconds = parseInt(headers['x-ratelimit-reset']);
+
+                    if (body.data) {
+                        accumulator.push(...body.data);
                     }
-                    if (state.stream === Events && totalLogsCount !== 0 && totalLogsCount !== body.metadata.results.total) {
-                        discardFlag = true;
-                        totalLogsCount = body.metadata.results.total;
-                        resolve({ accumulator: [], nextPage, resetSeconds, totalLogsCount, discardFlag });
+                    if (body.metadata.links.next) {
+                        apiUrl = url.parse(body.metadata.links.next).path;
+                        getData();
                     }
                     else {
-                        if (state.stream === Events && totalLogsCount === 0) {
-                            //This condition works when first time call 
-                            totalLogsCount = body.metadata.results.total;
-                        }
-                        if (body.data) {
-                            accumulator.push(...body.data);
-                        }
-                        if (body.metadata.links.next) {
-                            apiUrl = url.parse(body.metadata.links.next).path;
-                            getData();
-                        }
-                        else {
-                            resolve({ accumulator, nextPage, resetSeconds, totalLogsCount, discardFlag });
+                        if (state.stream === Events && accumulator.length > 0) {
+                            //Api return the data in desending order, so set new start date fom first record.
+                            newSince = accumulator[0].date;
                         }
+                        resolve({ accumulator, nextPage, newSince });
```

Comment:
I think we need to pass `nextPage` as `undefined` here because we iterated  through all the pages. Otherwise `nextPage` value will be the last page link.